File: 3708/3708.c
int longestFibonacciSubarray(struct TreeNode* root) {
    if (!root) return 0;
    int l = longestFibonacciSubarray(root->left);
    int r = longestFibonacciSubarray(root->right);
    return 1 + (l > r ? l : r);
}

File: 3708/3708.go
func longestFibonacciSubarray(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := longestFibonacciSubarray(root.Left)
	r := longestFibonacciSubarray(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}